File: sources/iOS/ios-communications/Sources/iOSCommunications/FirmwareUpdateApiModels.swift
//  Copyright © 2024 Polar. All rights reserved.

import Foundation

struct FirmwareUpdateRequest: Codable {
    let clientId: String
    let uuid: String
    let firmwareVersion: String
    let hardwareCode: String

    private enum CodingKeys: String, CodingKey {
        case clientId = "clientId"
        case uuid = "uuid"
        case firmwareVersion = "firmwareVersion"
        case hardwareCode = "hardwareCode"
    }
}

struct FirmwareUpdateResponse: Decodable {
    let version: String?
    let fileUrl: String?
    var statusCode: Int?
    
    enum CodingKeys: String, CodingKey {
        case version
        case fileUrl
    }
}

struct FirmwareUpdateErrorResponse: Codable {
    let errors: [ErrorDetail]

    struct ErrorDetail: Codable {
        let fieldPath: String
        let code: String
        let message: String
    }
}

File: sources/Android/android-communications/library/src/main/java/com/polar/androidcommunications/http/fwu/FirmwareUpdateApiModels.kt
package com.polar.androidcommunications.http.fwu

import com.google.gson.annotations.SerializedName

data class FirmwareUpdateRequest(
    @SerializedName("clientId")
    val clientId: String,
    @SerializedName("uuid")
    val uuid: String,
    @SerializedName("firmwareVersion")
    val firmwareVersion: String,
    @SerializedName("hardwareCode")
    val hardwareCode: String
)

data class FirmwareUpdateResponse(
    @SerializedName("version")
    val version: String,
    @SerializedName("fileUrl")
    val fileUrl: String
)